### ISSUE TITLE
Top Up Balance w/ MoonPay

### ISSUE THREAD

**Issue by corwinharrell** [near/near-wallet#1537]
### Summary
Today, users can add funds to their wallet by opening their exchange account, and go through the process of buying tokens and transferring them.

With this feature, users are enabled to use Moonpay to fund their existing account - repeating the process that before was available only at the account creation.

### Intended Outcome
As a user, I want to add funds to my wallet balance without leaving NEAR wallet user experience.

The success of this feature will be measured against the number of transactions performed within the application.

### How will it work?
Similarly to #5, users have a way to add funds to their balance, by using their credit card and the Moonpay buy process.
As a nice-to-have, we may enable this functionality via deep-linking, such that third-party applications can call this feature from their user interface.

**Comment by corwinharrell** [comment]
@vgrichina would love any feedback you might have for this. I believe this direction is more scalable than just adding a "buy with moonpay" button on the dashboard, since it allows us to point to exchanges as well as future integrations with other purchase methods (eg I know Metamask uses https://pay.sendwyre.com/ which I'm also curious if that is an option for us). I'd like to also leverage this pattern for the funding flow cc @Patrick1904.

![Purchase Methods](https://user-images.githubusercontent.com/5934188/112494567-8b15ce80-8d59-11eb-9b4b-613974fffb30.gif)

**Comment by Patrick1904** [comment]
@vgrichina @corwinharrell @jimmy3dita 

Since https://github.com/near/near-wallet/issues/1538 is somewhat blocked by https://github.com/near/near-wallet/pull/1556 - any objections to prioritize this issue before https://github.com/near/near-wallet/issues/1538? I could pick this up next.

**Comment by corwinharrell** [comment]
> @vgrichina @corwinharrell @jimmy3dita
> 
> Since #1538 is somewhat blocked by #1556 - any objections to prioritize this issue before #1538? I could pick this up next.

Not opposed at all but will wait on others to weigh in.
